core/src/mender-client.c

```diff
@@ -1203,6 +1203,40 @@ mender_check_artifact_requirements(mender_artifact_ctx_t *mender_artifact_ctx, m
 }
 #endif /* CONFIG_MENDER_FULL_PARSE_ARTIFACT */
 
+static mender_err_t
+mender_client_check_deployment(mender_api_deployment_data_t *deployment) {
+
+    mender_log_info("Checking for deployment...");
+    if (MENDER_OK != mender_api_check_for_deployment(deployment)) {
+        mender_log_error("Unable to check for deployment");
+        return MENDER_FAIL;
+    }
+
+    /* Check if deployment is available */
+    if ((NULL == deployment->id) || (NULL == deployment->artifact_name) || (NULL == deployment->uri) || (NULL == deployment->device_types_compatible)) {
+        mender_log_info("No deployment available");
+        return MENDER_DONE;
+    }
+
+    /* Reset flags */
+    mender_client_deployment_needs_set_pending_image = false;
+
+    /* Create deployment data */
+    if (NULL != mender_client_deployment_data) {
+        mender_log_warning("Unexpected stale deployment data");
+        cJSON_Delete(mender_client_deployment_data);
+    }
+    if (NULL == (mender_client_deployment_data = cJSON_CreateObject())) {
+        mender_log_error("Unable to allocate memory");
+        return MENDER_FAIL;
+    }
+    cJSON_AddStringToObject(mender_client_deployment_data, "id", deployment->id);
```

Comment:
Unrelated to this PR, but we should probably check the return value of these function calls as they return `NULL` on error. Feel free to fix it if you want :)